src/web/models.py

```diff
@@ -6,6 +6,15 @@
 from handyhelpers.models import HandyHelperBaseModel
 
 
+class Tag(HandyHelperBaseModel):
+    """A Tag that describes attributes of a Event."""
+
+    value = models.CharField(max_length=32, unique=True, null=False)
+
+    def __str__(self) -> str:
+        return self.value
```

Comment:
Would these ever also be used to describe a TechGroup? Like if we tag various groups as more and more groups are added. If that's the case, we may want to use generic foreign keys: https://docs.djangoproject.com/en/5.0/ref/contrib/contenttypes/

Reply:
The `ManyToMany` relation I'm using now can be used to relate (the same) `Tag`s to `TechGroup`s too.

```
sqlite> pragma table_info(web_event_tags);
cid  name      type     notnull  dflt_value  pk
---  --------  -------  -------  ----------  --
0    id        INTEGER  1                    1 
1    event_id  bigint   1                    0 
2    tag_id    bigint   1                    0 

 
sqlite> pragma table_info(web_techgroup_tags);
cid  name          type     notnull  dflt_value  pk
---  ------------  -------  -------  ----------  --
0    id            INTEGER  1                    1 
1    techgroup_id  bigint   1                    0 
2    tag_id        bigint   1                    0 
```

I'll take a closer look at generic foreign keys, but at a quick glance I don't see a reason to use them over the `ManyToMany` relation in this use-case. I'm also not sure they'd provide the nice selection box that we get now 
![image](https://github.com/SpokaneTech/SpokaneTech_Py/assets/17833726/226147c9-39de-4e56-a7bc-d8fc7297803f).

Reply:
I think `ManyToMany` works well. The main advantage to generic foreign keys is that if there were a bunch of different models that we wanted to have tags, instead of adding `ManyToMany` fields to every model, you can just configure the Tag model to be able to be linked to an instance of any model. I think sticking what you have right now is good.